code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/AbstractTokenUtil.java

```diff
@@ -396,4 +397,49 @@ public static boolean isUnrestrictedAccess(String accessMode){
         return UNRESTRICTED_ACCESSMODE.equalsIgnoreCase(accessMode);
     }
 
+    public Identity jsonToIdentity(Map<String, Object> jsonData) {
+        String externalId = ObjectUtils.toString(jsonData.get("externalId"));
+        String externalIdType = ObjectUtils.toString(jsonData.get("externalIdType"));
+        String name = ObjectUtils.toString(jsonData.get("name"));
+        String profilePicture = ObjectUtils.toString(jsonData.get("profilePicture"));
+        String profileUrl = ObjectUtils.toString(jsonData.get("profileUrl"));
+        String login = ObjectUtils.toString(jsonData.get("login"));
+        return new Identity(externalIdType, externalId, name, profileUrl, profilePicture, login);
+    }
+
+    public Token getUserIdentityFromJWT() {
+        Token token = new Token();
+        Map<String, Object> jsonData = getJsonData();
+        if (jsonData == null) {
+            return null;
+        }
+        AuthToken authToken= authTokenDao.getTokenByKey(getJWT());
+        String tokenId = (String)ApiContext.getContext().getIdFormatter().formatId(objectManager.getType(Account.class), authToken.getId());
```

Comment:
The formatId is being called by passing "type" = Account.class, this needs to be corrected to use the correct type